rebac-admin-backend/v1/identities.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// GetIdentities returns the list of known identities.
+// (GET /identities)
+func (h handler) GetIdentities(w http.ResponseWriter, req *http.Request, params resources.GetIdentitiesParams) {
+	identities, err := h.Identities.ListIdentities(&params)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	response := resources.GetIdentitiesResponse{
+		Data:   identities.Data,
+		Status: 200,
+	}
+
+	writeResponse(w, 200, response)
+}
+
+// PostIdentities adds a new identity.
+// (POST /identities)
+func (h handler) PostIdentities(w http.ResponseWriter, req *http.Request) {
+	identity := new(resources.Identity)
+	defer req.Body.Close()
+
+	err := json.NewDecoder(req.Body).Decode(identity)
+	if err != nil {
+		writeErrorResponse(w, NewValidationError("Request doesn't match the expected schema"))
+		return
+	}
+
+	identity, err = h.Identities.CreateIdentity(identity)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 201, identity)
+}
+
+// DeleteIdentitiesItem deletes the specified identity.
+// (DELETE /identities/{id})
+func (h handler) DeleteIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	_, err := h.Identities.DeleteIdentity(id)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	w.WriteHeader(200)
+}
+
+// GetIdentitiesItem returns the identity identified by the provided ID.
+// (GET /identities/{id})
+func (h handler) GetIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	identity, err := h.Identities.GetIdentity(id)
+	if err != nil {
+		response := h.IdentitiesErrorMapper.MapError(err)
+		writeResponse(w, response.Status, response)
+		return
+	}
+
+	writeResponse(w, 200, identity)
+}
+
+// PutIdentitiesItem updates the identity identified by the provided ID.
+// (PUT /identities/{id})
+func (h handler) PutIdentitiesItem(w http.ResponseWriter, req *http.Request, id string) {
+	identity := new(resources.Identity)
+	defer req.Body.Close()
+
+	err := json.NewDecoder(req.Body).Decode(identity)
```

Comment:
Limit the `err` scope to the `if`-statement. Won't comment on the next occurrences.